OOP/hw_1.py

```diff
@@ -0,0 +1,429 @@
+# -*- coding: utf-8 -*-
+"""HW_1.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1duyXA-btTf7KsFWM2P6rgrBECtgK9pwq
+
+# Задание 1 (5 баллов)
+
+Напишите классы **Chat**, **Message** и **User**. Они должны соответствовать следующим требованиям:
+
+**Chat**:
++ Должен иметь атрибут `chat_history`, где будут храниться все сообщения (`Message`) в обратном хронологическом порядке (сначала новые, затем старые)
++ Должен иметь метод `show_last_message`, выводящий на экран информацию о последнем сообщении
++ Должен иметь метод `get_history_from_time_period`, который принимает два опциональных аргумента (даты с которой и по какую мы ищем сообщения и выдаём их). Метод также должен возвращать объект типа `Chat`
++ Должен иметь метод `show_chat`, выводящий на экран все сообщения (каждое сообщение в таком же виде как и `show_last_message`, но с разделителем между ними)
++ Должен иметь метод `recieve`, который будет принимать сообщение и добавлять его в чат
+
+**Message**:
++ Должен иметь три обязательных атрибута
+    + `text` - текст сообщения
+    + `datetime` - дата и время сообщения (встроенный модуль datetime вам в помощь). Важно! Это должна быть не дата создания сообщения, а дата его попадания в чат! 
+    + `user` - информация о пользователе, который оставил сообщение (какой тип данных использовать здесь, разберётесь сами)
++ Должен иметь метод `show`, который печатает или возвращает информацию о сообщении с необходимой информацией (дата, время, юзер, текст)
++ Должен иметь метод `send`, который будет отправлять сообщение в чат
+
+**User**:
++ Класс с информацией о юзере, наполнение для этого класса придумайте сами
+
+Напишите несколько примеров использования кода, которое показывает взаимодействие между объектами.
+
+В тексте задания намерено не указано, какие аргументы должны принимать методы, пускай вам в этом поможет здравый смысл)
+
+В этом задании не стоит флексить всякими продвинутыми штуками, для этого есть последующие
+
+В этом задании можно использовать только модуль `datetime`
+"""
+
+from datetime import datetime
+
+
+class Chat:
+    # Ваш код здесь
+    
+    def __init__(self):
+        self.chat_history = []
+    
+    def _recieve(self, message):
+        if isinstance(message, Message):
+            self.chat_history.insert(0, message)
+        else:
+            print('This is not message')
+          
+    def _new_chat_from_mlist(self, chat_list):
+        chat = type(self)()
+        chat.chat_history = chat_list
```

Comment:
Мне кажется, что удобно было бы довавить в конструктор аргумент для того, чтобы инициализировать сразу непустой чат, тогда тут было бы что-то вот такое
```suggestion
        chat = type(self)(chat_list)
```